airflow/models/xcom.py

```diff
@@ -161,27 +183,6 @@ def set(
         if dag_run_id is None:
             raise ValueError(f"DAG run not found on DAG {dag_id!r} with ID {run_id!r}")
 
-        # Seamlessly resolve LazySelectSequence to a list. This intends to work
-        # as a "lazy list" to avoid pulling a ton of XComs unnecessarily, but if
-        # it's pushed into XCom, the user should be aware of the performance
-        # implications, and this avoids leaking the implementation detail.
-        if isinstance(value, LazySelectSequence):
```

Comment:
We can probably delete that class now too, as I think this was the last usage of it?

Reply:
Its actually used in task instance models xcom_push and xcom_pull. Hence I didnt wipe it right away